src/main/java/com/fulcrumgenomics/util/DemuxFastqs.java

```diff
@@ -677,14 +697,24 @@ void convertQuality(final byte[] quals, final FastqQualityFormat version) {
             return sampleBarcodeQualities;
         }
 
-        private String getMolecularBarcode(final List<FastqRecord>  records) {
-            final byte[][] molecularBarcode = new byte[records.size()][];
+        private byte[][] getMolecularBarcodes(final List<FastqRecord> records) {
+            final byte[][] molecularBarcodes = new byte[records.size()][];
+            int numNonEmptyMolecularBarcodes = 0;
             for (int i = 0; i < records.size(); i++) {
                 final FastqRecord record = records.get(i);
                 final ReadStructureInfo readStructureInfo = readStructureInfos.get(i);
-                molecularBarcode[i] = getBasesAtCycles(record.getReadString().getBytes(), readStructureInfo.molecularBarcodeCycles);
+                molecularBarcodes[i] = getBasesAtCycles(record.getReadString().getBytes(), readStructureInfo.molecularBarcodeCycles);
```

Comment:
Also not related to this change, but does this mean that if I have a read structure of 8M8B8M76T that I'll get the 16 molecular barcode bases back as one UMI read not two?